src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.FormatDocument.cs

```diff
@@ -1,27 +1,57 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Implementation.CodeCleanup;
+using Microsoft.CodeAnalysis.Editor.Shared.Options;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Extensions;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 using Microsoft.CodeAnalysis.Text;
 using Microsoft.VisualStudio.Commanding;
+using Microsoft.VisualStudio.Text.Editor;
 using Microsoft.VisualStudio.Text.Editor.Commanding.Commands;
 using VSCommanding = Microsoft.VisualStudio.Commanding;
 
 namespace Microsoft.CodeAnalysis.Editor.Implementation.Formatting
 {
-    internal partial class FormatCommandHandler
+    internal partial class FormatCommandHandler : ForegroundThreadAffinitizedObject
     {
+        private bool _infoBarOpen = false;
```

Comment:
making this stateful seems very dangerous.  Why can we not just query to find out of the info bar is open instead of storing this as state in then handler?

Reply:
I am not sure which query it is about.  We only get the IInfoBarService when the bool value is false and it is consistent with elsewhere in our code.

Reply:
@JieCarolHu I think we need to save this in option so that user see this only once rather than every time user open new VS.

something like this
http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/Experimentation/AnalyzerVsixSuggestedActionCallback.cs,196

Reply:
+1 on this suggestion. It would be annoying to keep seeing the info bar every time.

---
In reply to: [192850157](https://github.com/dotnet/roslyn/pull/26802#discussion_r192850157) [](ancestors = 192850157)